### DISCUSSION TITLE
WPFUI collaboration for Win11 Theming

### DISCUSSION THREAD

**Discussion by pchaurasia14** [dotnet/wpf#8533] [Announcements]
We are excited to share another development in our Win11 theming effort. After careful evaluation of various options to deliver Win11 theming for WPF repository, we have decided to join forces with [WPFUI](https://github.com/lepoco/wpfui) for this initiative.

Here's a glimpse into our evaluation journey:

- Evaluation Process:
      Our primary goal was to avoid reinventing the wheel and instead leverage the strength of the open source community.
      We assessed multiple open source projects. Our focus was on themes closely aligned with WinUI design guidelines.

- Assessment:
      We found that WPFUI, particularly championed by @pomianowski, closely mirrors the Win11 theming designs we aim to provide. Interestingly, WPFUI is already a crucial part of [PowerToys](https://github.com/microsoft/PowerToys), further solidifying its reliability and compatibility.

- Collaboration with lepoco owner:
      We extend our heartfelt gratitude to @pomianowski for agreeing to collaborate with us on this feature.

- Upcoming Milestones:
      We will be publishing detailed milestones outlining our plans for Win11 Theming in a separate issue.

We appreciate your ongoing support and enthusiasm as we continue to evolve and improve WPF. Together, we look forward to delivering Win11 theming that surpasses expectations.

**Comment by ghost1372** [comment]
all wpf fluent library is a fork from original repo https://github.com/Kinnara/ModernWpf and this repo Recently has been activated and it is exactly the ported version of WinUI 3.
Also, when I tested [Lipoco](https://github.com/lepoco/wpfui) a few weeks ago, it was full of bugs and even some controls and styles are not according to the original control/style! for example you can try NavigationView control

**Reply by Nice3point** [reply]
ModernWPF looks a bit outdated. It doesn't matter who ported whom, we need to achieve an amazing WPF look out of the box

**Reply by niels9001** [reply]
ModernWPF was using styles that were used in UWP. With WinUI 2.5/2.6 those styles changed and but ModernWPF never adopted those.  This effort would bring WinUI 2.8 styling to WPF, making apps look at home on Windows 11.

**Reply by ghost1372** [reply]
@Nice3point yes readme is not updated, and developer is working for winui 2.8 styles, for me It matters who ported it. I prefer to use a real ported version instead of a fake version. @niels9001 they are working on it.

**Reply by BlameTwo** [reply]
@niels9001  To be honest, I think the style and details of WPFUI are not even as good as Modern WPF. Just from the buttons, it can be seen that WPFUI had a lot of animation details in the early days, but later on, for some reasons, it may have cut off a large part of the animation.

**Reply by Difegue** [reply]
ModernWpf was dead for months, even if they're finishing the 2.8 port now that ship has sailed and everyone moved to WPFUI in the meantime despite the fact it wasn't a 1:1 port ¯\\\_(ツ)_/¯

**Comment by ali50m** [comment]
Is it able to make the WPF style in a standlone nuget or something similar, then you can reference it as the default style provider? There is rumor says Win12 will come out in 2024, it may impact the influence of the WPF Win11 Theming.

**Reply by BoyFaceGirl** [reply]
Haha, they have been discussing the Win11 theme for a year and it hasn't started yet. It's almost 2024. They quickly updated their future plans to port WPFUI, which they jokingly said. It's estimated that the port will only be completed by 2030. When the Windows 20 interview comes, they may start discussing the Win12 theme again. The endless anticipation will always bring disappointment. I am eagerly anticipating Avalonia now.

**Comment by BoyFaceGirl** [comment]
WPFUI is full of various bugs, especially the NavigateView control, which have persisted for several years without being fixed. It's better to pay attention to UI WPF Modern, the styles and usage of all controls are basically synchronized with Winui3, and the documentation of Winui3 is also applicable to UI.WPF.Modern

https://github.com/inkorestudios/ui.wpf.modern

**Reply by BartoszCichecki** [reply]
"several years" is a bit of a stretch when first release of wpfui was in March '22.

**Comment by lindexi** [comment]
Though WPFUI is full of various bugs, I am still very pleased to see the WPF repository embracing the WPFUI repository, and I hope it can do even better. I will also do my best to contribute my own efforts.

Fixing bugs is something developers are good at.

**Reply by BoyFaceGirl** [reply]
大佬, 
I am also Chinese. I am surprised that you have made so much contribution to WPF. Can you publish a special blog on WPF using WinUI3? I have seen some scattered problem-solving solutions you have written, and I look forward to organizing them into one, especially the dependencies between WPF, Island, and Windows app SDK, as well as the configuration in the. csproj file

**Reply by lindexi** [reply]
@BoyFaceGirl See:

- https://blog.lindexi.com/post/WPF-%E4%B8%8D%E5%AE%89%E8%A3%85-WindowsAppSDK-%E4%BD%BF%E7%94%A8-WinRT-%E5%8A%9F%E8%83%BD%E7%9A%84%E6%96%B9%E6%B3%95.html
- https://blog.lindexi.com/post/WPF-%E9%80%9A%E8%BF%87-WindowsAppSDK-%E4%BD%BF%E7%94%A8-WinRT-%E7%9A%84%E6%89%8B%E5%86%99%E8%AF%86%E5%88%AB%E5%8A%9F%E8%83%BD.html


For more, see https://blog.lindexi.com/post/%E5%8D%9A%E5%AE%A2%E5%AF%BC%E8%88%AA.html

**Comment by BlameTwo** [comment]
wish you success🤣

**Comment by Baka632** [comment]
Something meaningful starts, after all.

**Comment by GalaxyGaming2000** [comment]
Will We Still be able to use this on windows 10?

**Comment by NeverMorewd** [comment]
More or less. It has been started

**Reply by get-flat** [reply]
Could you please consider decoupling UI layer (similarly to CommunityToolkit). It seems that by default WPFUI forces ViewModels to be dependent on UI framework to implement navigation (i.e. INavigationAware etc)

**Comment by otomad** [comment]
Have you tried [UI.WPF.Modern](https://github.com/iNKORE-NET/UI.WPF.Modern)? It seems to handle the styling and details better than [WPFUI](https://github.com/lepoco/wpfui) does in some case.

**Comment by xmaxrayx** [comment]
my frustration is I cant do Arcylic style with corners window... maybe GDI only option but it's not that smooth.
